### ISSUE TITLE
Unable to add a Component to a Dynamic Zone after 4.15.1 upgrade

### ISSUE THREAD

**Issue by dennism501** [strapi/strapi#18665]
### Required System information

- Node.js version: 20.9.0
- NPM version: 10.1.0
- Strapi version: 4.15.1
- Database: SQLite
- Operating system: Mac OS
- Is your project Javascript or Typescript: JS

### Describe the bug

After upgrading to 4.15.0 and 4.15.1 as well, I am unable to add a component to a Dynamic zone. Whenever you try to add a component to a DZ, the page breaks and you can not procede 

### Steps to reproduce the behavior

1. Update `package.json` to Strapi 4.15.*
2. delete `node_modules, yarn.lock, .cache` 
3. Run `yarn && yarn build && yarn develop` 
4. Go to any content type with a Dynamic zone in the Content Type Builder
5. Click on add component button
6. See error 

### Expected behavior

Being able to add a component to a Dynamic Zone in the content type builder

### Screenshots

I created this [loom video](https://www.loom.com/share/f0b82a6ecc8b47a4be8c4f4131e6abe5?sid=b2595bd9-1161-4a0c-b5a0-ecc7ab22349b) to show the steps of recreation. 

<img width="1505" alt="image" src="https://github.com/strapi/strapi/assets/15628660/ea6804c1-dcfa-4cb3-9748-e5f52936976e">

### Code snippets

If applicable, add code samples to help explain your problem.

**Comment by dennism501** [comment]
Escalation with TID: 5147

**Comment by Ceyapps** [comment]
i'm having the same issue

**Comment by dgrebb** [comment]
Also impacting a big project for me.

I've tried downgrading back to `4.14.2`, which also has this issue, as well as each patch/minor version up to and including `4.15.2`, which is a no-go as `--legacy-peer-deps` requires changes across CI (and the issue persists).

I get a different error, but it's in the same reproducible step @dennism501 shared. Clicking the "Add Component" button throws:

```javascript
Uncaught TypeError: r is undefined
    xa http://localhost:1337/admin/content-type-builder.2d053d33.chunk.js:149
```

Database corruption? Making manual changes to the schema files has been OK, but I'm blocked on a large, dynamic-component heavy feature.

**Comment by dgrebb** [comment]
I had an empty `4.14.0` install — for all intents and purposes fresh and clean.

Copied over `api` and `components`, fired it up, and everything is working, including "Add Component", through to the publish of content.

I'll take a deeper look into whether this might be directly related to using postgres, which is in use in the envs where this error is popping up (for me).

@dennism501 @Ceyapps what DBs are you using?

**Comment by derrickmehaffy** [comment]
Raised to critical based on internal discussion with DX Squad. CC @marcoautiero

**Comment by dgrebb** [comment]
Following the comments on the Community Forum, downgrading to `4.13.1` worked! `4.13.0` is [specifically mentioned in the forum](https://forum.strapi.io/t/error-when-clicking-on-add-a-component-in-dynamic-zone-typeerror-cannot-read-properties-of-undefined-reading-length/33211/4), but `4.13.1` works for me 👍.

No data loss, all fields and content intact.

Heads up @remidej @Ceyapps @dennism501 — hope this helps. Remember to `npm run build` 😄 (I didn't).

**Comment by dgrebb** [comment]
In hopes to help with triage, `4.15.4` also has this issue:

```javascript
TypeError: r is undefined
    qn http://localhost:1337/admin/7743.9831df1a.chunk.js:151
```

**Comment by EugeneDraitsev** [comment]
It seems like in `4.15.4`  the error occurs in `packages/core/content-type-builder/admin/src/components/SelectComponents/index.js` when trying to access `value.length`:

```jsx
  const displayedValue = formatMessage(
    {
      id: getTrad('components.SelectComponents.displayed-value'),
      defaultMessage:
        '{number, plural, =0 {# components} one {# component} other {# components}} selected',
    },
    { number: value.length } // <- Error is here
  );
```

If I edit it as `{ number: value?.length }` in the `/dist/build` folder, the Dynamic zone works as expected. 
I'll try to create a PR with this small fix

**Comment by derrickmehaffy** [comment]
Could I have a few of you try upgrading to https://github.com/strapi/strapi/releases/tag/v4.15.5-alpha.2

To test if you still run into the issue?

**Comment by derrickmehaffy** [comment]
> @derrickmehaffy still fails in `4.15.5-alpha.2`

Thanks for testing

**Comment by CodeStarrk** [comment]
This is critical for me, can I downgrade to a version to get this to work?

**Comment by dgrebb** [comment]
@CodeStarrk [this thread](https://forum.strapi.io/t/error-when-clicking-on-add-a-component-in-dynamic-zone-typeerror-cannot-read-properties-of-undefined-reading-length/33211/4) helped me.

### tl;dr

1. update various package versions to `4.14.0`
2. `npm install` or, `rm -rf node_modules package-lock.json && npm i`
3. `npm run build`
4. `npm run develop`

I've [been running](https://github.com/strapi/strapi/issues/18665#issuecomment-1806531062) `4.14.0` for ~5 days now without issue, and built out a complex system of singles, collections, components, relations, and dynamic zones without error.

**Comment by glani** [comment]
Checked 4.15.5-alpha.4 does not work neither

**Comment by Apostata** [comment]
Thank you @glani. works for me!

**Comment by FlorianLauterbach** [comment]
for the guys that are new (like me): 
4.14.6 is working for me, too.. but first I had to run a `yarn build` before the `yarn develop` is working again

**Comment by joshuaellis** [comment]
Resolved by #18808

**Comment by maleksandrov** [comment]
I still need help with even the latest alpha version.
![image](https://github.com/strapi/strapi/assets/10142018/dc50d25b-7851-4a17-9e8a-eee65d09cb8d)

Local URL: http://localhost:1337/admin/plugins/content-type-builder/content-types/api::page.page
The error appears once I click on Add a component:
![image](https://github.com/strapi/strapi/assets/10142018/06360263-7408-4e13-ae91-de8f67f45dab)

![image](https://github.com/strapi/strapi/assets/10142018/ef8ba6b6-f54e-4e53-a8d9-717e44ad3aee)

@joshuaellis, please share more details on how to handle this! If the issue is already fixed! 
I'm using this git repo - https://github.com/strapi/nextjs-corporate-starter. 

Q: Under which version does this functionality work? 

Thank you for any advice and support!

**Comment by remidej** [comment]
Hey @maleksandrov, alpha.4 was released 6 days ago, so it's expected that it doesn't have the fix yet. It will come in the next alpha, which I hope will come soon

**Comment by sunil1199** [comment]
Hi there,
When I click on 'Add a component' then white screen appears, I have created with npm.
please help
![image_2023_11_22T05_47_13_628Z](https://github.com/strapi/strapi/assets/124344290/c66bf85c-d43f-4f62-b794-e364f082ceaa)
![image_2023_11_22T05_41_16_216Z](https://github.com/strapi/strapi/assets/124344290/5aa81f23-7152-49eb-a50d-db93e3f69869)

**Comment by IGernhardt** [comment]
Also having the same problem here in version 4.15.4. Will try to downgrade now

**Comment by sunil1199** [comment]
@IGernhardt can you please know the step of downgrading from 4.15.4 to 4.14.0

**Comment by IGernhardt** [comment]
> @IGernhardt can you please know the step of downgrading from 4.15.4 to 4.14.0

I downgraded by doing the following:

1. changing the version numbers in the package.json (deleting @strapi/plugin-cloud cause I didn't need it and there's no 4.14.0 version of that specific package).
2. running >npm install 
3. running >npm build

Hope this helps

**Comment by sunil1199** [comment]
> > @IGernhardt can you please know the step of downgrading from 4.15.4 to 4.14.0
> 
> I downgraded by doing the following:
> 
> 1. changing the version numbers in the package.json (deleting @strapi/plugin-cloud cause I didn't need it and there's no 4.14.0 version of that specific package).
> 2. running >npm install
> 3. running >npm build
> 
> Hope this helps

it's worked thank you so much :)

**Comment by remidej** [comment]
Hello, could you all try version `4.15.5-alpha.5`? It should contain the fix for this issue

**Comment by Apostata** [comment]
Hi @remidej , version 4.15.5-alpha.5 working! Thank you!

**Comment by strlrd-29** [comment]
Upgrading to 4.15.5 fixed it to me.

**Comment by andreasrein** [comment]
Perfect, 4.15.5 solved it! Thanks!
